js.go

```diff
@@ -2751,10 +2753,18 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 					// wait this time.
 					noWait = false
 					err = sendReq()
-				} else if err == ErrTimeout && len(msgs) == 0 {
+				} else if err == ErrTimeout {
 					// If we get a 408, we will bail if we already collected some
-					// messages, otherwise ignore and go back calling NextMsg.
-					err = nil
+					// messages, otherwise ignore and go back calling nextMsg.
+					if len(msgs) == 0 {
```

Comment:
As suggested above, I don't think we need that with unique request subject suffix tokens.

Reply:
Yeah, I left it just to make absolutely sure it had no other use.